runtime/etc/mli/mli_common_code.c

```diff
@@ -190,6 +191,36 @@ int chpl_mli_pull(void* socket, void* buffer, size_t bytes, int flags) {
   return zmq_recv(socket, buffer, bytes, flags);
 }
 
+// Determine connection information for that socket.
+static
+char * chpl_mli_connection_info(void* socket) {
+  // Determine port used by ZMQ for socket.
+  size_t lenPort = 256;
+  char* portRes = (char *)mli_malloc(lenPort);
+  int portErr = zmq_getsockopt(socket, ZMQ_LAST_ENDPOINT,
+                               portRes, &lenPort);
+
+  // Get to port portion of the last endpoint.
+  char *traveler = strchr(portRes, ':');
+  traveler = strchr(traveler + 1, ':');
+  traveler++;
+
+  // Determine hostname of where we are currently running
+  size_t lenHostname = 256;
+  char* hostRes = (char *)mli_malloc(lenHostname);
+  err_t hostErr = gethostname(hostRes, lenHostname);
+
+  // Recreate the connection using the hostname instead of 0.0.0.0
+  char* fullConnection = (char *)mli_malloc(lenHostname + lenPort);
```

Comment:
Hmmmmm, you do casts from `malloc()` in your C code but I don't. You technically don't need them, but I wonder if there is an upside (IE, if the code is ever compiled by a C++ compiler).

Reply:
chpl_malloc technically returns `void *`.  It seems like general practice in the code I've seen involving malloc online to do a cast to the appropriate type, but that may very well be superstitious on my part